client/src/send_and_confirm_transactions_in_parallel.rs

```diff
@@ -477,23 +473,33 @@ pub async fn send_and_confirm_transactions_in_parallel<T: Signers + ?Sized>(
         // clear the map so that we can start resending
         unconfirmed_transasction_map.clear();
 
-        sign_all_messages_and_send(
-            &progress_bar,
-            &rpc_client,
-            &tpu_client,
-            messages_with_index,
-            signers,
-            &context,
-        )
-        .await?;
-
-        // wait until all the transactions are confirmed or expired
-        confirm_transactions_till_block_height_and_resend_unexpired_transaction_over_tpu(
-            &progress_bar,
-            &tpu_client,
-            &context,
-        )
-        .await;
+        let futures = [
+            sign_all_messages_and_send(
+                &progress_bar,
+                &rpc_client,
+                &tpu_client,
+                messages_with_index,
+                signers,
+                &context,
+            )
+            .boxed_local(),
+            async {
+                // Give the signing and sending a head start before trying to
+                // confirm and resend
+                tokio::time::sleep(TPU_RESEND_REFRESH_RATE).await;
+                confirm_transactions_till_block_height_and_resend_unexpired_transaction_over_tpu(
+                    &progress_bar,
+                    &tpu_client,
+                    &context,
+                )
+                .await;
+                // Infallible, but required to have the same return type as
+                // `sign_all_messages_and_send`
+                Ok(())
+            }
+            .boxed_local(),
+        ];
+        join_all(futures).await.into_iter().collect::<Result<_>>()?;
```

Comment:
I've noticed that sometimes... do you have any idea what causes it? In my experience, I don't see the block height updating, which makes me think that something in `confirm_transactions_till_block_height_...` is hanging.

I'd prefer that we only abort when we know that the block height has been passed, rather than hardcoding 2 minutes. We can detach the future with `tokio::spawn`, loop checking `is_finished()` on it, and abort it if the blockhash is expired. What do you think?

Reply:
Sweet, do you mind if I include it in a separate PR? It isn't directly related to the changes here and can be done independently

Reply:
Nah not at all